### PR TITLE
Lumen support and some bugfixes

### PR DESCRIPTION
**Pull request by mpetrunic**
Hi,
I'm trying to use this with lumen 5.5.

Compatibility issue:
- lumen doesn't support config publish

Bugs:
- got timer not started because get duration in millis is called before stop on transaction
- replace `$request->route()->uri` with `$request->path()` because it's supported in both lumen and laravel

cheers

### CODE REVIEW

**Comment by philkra** [comment]
great, thanks for your contribution @mpetrunic . But we'll have to check first if this doesn't break the laravel portage.

**Comment by dstepe** [comment]
@mpetrunic Looks like maybe a rebase is needed. I see the `philkra/elastic-apm-php-agent` package version is dropped back down to 6.5.2 and we lose the asserted duration when the transaction is stopped. Those were merged in not long ago.

**Comment by mpetrunic** [comment]
@dstepe oh, I've drpped it because agent isn't published yet. But this package doesn't use functionality that was added in`philkra/elastic-apm-php-agent@6.5.3`. Composer will install anyway latest version once it's published :)

**Comment by mpetrunic** [comment]
If you all agree, I can revert it back to forcing 6.5.3 version

**Comment by dstepe** [comment]
6.5.3 is required due to changes in the provider to use the Timer class from that package. That release updates the Timer class to accept a start time rather automatically starting with the current time. The provider has been updated to create an instance of the Timer initialized with the REQUEST_TIME_FLOAT.

https://github.com/philkra/elastic-apm-laravel/blob/master/src/Providers/ElasticApmServiceProvider.php#L65

As you can see, we can't really drop below 6.5.3.

But I see what you mean. That release is not listed on Packagist yet. @philkra I've noticed other releases take a while to appear. How long do you expect that to take and is there anyway to tickle Packagist to update?

**Comment by mpetrunic** [comment]
@dstepe oh, missed that part :)

I think I managed to sort it all out to keep history clean. 
Meanwhile I've moved `mergeConfigFrom` to register method because, lumen doesn't have automatic config publish so there was no default config in register method if you do not copy config file manually.

**Comment by philkra** [comment]
@dstepe , that is odd with packagist! I'll check the webhook's

**Comment by philkra** [comment]
@dstepe , if you are fine with this PR, then it can be merged.

**Comment by dstepe** [comment]
How about a PR to the `development` branch? I'm not comfortable signing off on the Laravel provider changes by visual review only. I only know enough to make it work, and not necessarily correctly. I'd prefer to test this directly before it hits master.

**Comment by mpetrunic** [comment]
@dstepe done, didn't notice you had dev branch, sorry.

File: src/Middleware/RecordTransaction.php [hunk 1]
```diff
@@ -66,7 +66,7 @@ public function handle($request, Closure $next)
             $transaction->setTransactionName($this->getRouteUriTransactionName($request));
         }
 
-        $transaction->stop($this->timer->getElapsedInMilliseconds());
```
Comment (dstepe) [inline src/Middleware/RecordTransaction.php]: The use of the Timer to get the transaction duration should be retained.

Comment (mpetrunic) [reply src/Middleware/RecordTransaction.php]: Huh, now I see what's wrong with agent version 6.5.2, it was missing timer constructor and it was throwing here timer not started. that's why I've removed it, but now it doesn't calculate duration from the time request came, but when it activated middleware?

Might be good to start tracking changes  of agent in some changelog though :)

**Pull request merged**